main.rb

```diff
@@ -0,0 +1,76 @@
+require 'open3'
+require 'os'
+
+def get_env_variable(key)
+  return (ENV[key] == nil || ENV[key] == "") ? nil : ENV[key]
+end
+
+def env_has_key(key)
+  value = get_env_variable(key)
+	return value unless value.nil? || value.empty?
+
+	abort("Input #{key} is missing.")
+end
+
+$output_path = env_has_key("AC_OUTPUT_DIR")
+$repo_path = env_has_key("AC_REPOSITORY_DIR")
+$detox_configuration = env_has_key("AC_RN_DETOX_CONFIGURATION")
+$detox_params = get_env_variable("AC_RN_DETOX_ARGS")
+
+
+$exit_status_code = 0
+def run_command(command, skip_abort)
+    puts "@@[command] #{command}"
+    status = nil
+    stdout_str = nil
+    stderr_str = nil
+
+    Open3.popen3(command) do |stdin, stdout, stderr, wait_thr|
+        stdout.each_line do |line|
+            puts line
+        end
+        stdout_str = stdout.read
+        stderr_str = stderr.read
+        status = wait_thr.value
+    end
+
+    unless status.success?
+        puts stderr_str
+        unless skip_abort
+            exit 1
+        end
+        $exit_status_code = 1
+    end
+end
+
+def runTests
+
+    if OS.mac?
+      run_command("brew tap wix/brew && brew install applesimutils", false)
+    end 
+
+    yarn_or_npm = "npm"
+    if File.file?("#{$repo_path}/yarn.lock")
+      yarn_or_npm = "yarn"
+    end
```

Comment:
IMO this usage more useful
```suggestion
      yarn_or_npm = File.file?("#{$repo_path}/yarn.lock") ? "yarn" : "npm"
```